app/assets/javascripts/select-kit/addon/components/composer-actions.js

```diff
@@ -30,30 +30,34 @@ export default DropdownSelectBoxComponent.extend({
   pluginApiIdentifiers: ["composer-actions"],
   classNames: ["composer-actions"],
   isEditing: equal("action", EDIT),
+  isInSlowMode: gt("topic.slow_mode_seconds", 0),
 
   selectKitOptions: {
     icon: "iconForComposerAction",
     filterable: false,
     showFullTitle: false,
   },
 
-  iconForComposerAction: computed("action", "whisper", "noBump", function () {
-    if (this.isEditing) {
+  @discourseComputed("isEditing", "action", "whisper", "noBump", "isInSlowMode")
+  iconForComposerAction(isEditing, action, whisper, noBump, isInSlowMode) {
```

Comment:
@jjaffeux a code smell again since there're so many arguments here. But it's harder to imagine fix here and I feel like it's better not to move so fast with the refactoring of composer since it's so important component.

Reply:
Yes no need to rush.

A possible solution would be to create a state object:

```
@discourseComputed("isEditing", "isInSlowMode")
composerState(isEditing, isInSlowMode) {
  return  { isEditing, isInSlowMode };
```

And then you could use it as property in iconForComposerAction:

```
@discourseComputed("composerState.{isEditing,isInSlowMode}", "action", "whisper", "noBump")
iconForComposerAction(composerState, action, whisper, noBump) {
  return ...
}
```

But yes no need to rush this in this PR, you can keep it that way, it might need more thought.

Reply:
👍 at first glance looks like a cool solution, but, yeah, better later.